src/Bridge/Doctrine/Orm/Filter/BooleanFilter.php

```diff
@@ -50,18 +54,31 @@ public function apply(QueryBuilder $queryBuilder, QueryNameGeneratorInterface $q
 
         $properties = $this->extractProperties($request);
 
-        foreach ($properties as $property => $boolean) {
-            if (!$this->isPropertyEnabled($property) || !$this->isPropertyMapped($property, $resourceClass)) {
+        foreach ($properties as $property => $value) {
+            if (
+                !$this->isPropertyEnabled($property) ||
+                !$this->isPropertyMapped($property, $resourceClass) ||
+                !$this->isBooleanField($property, $resourceClass)
+            ) {
                 continue;
             }
 
-            if (empty($boolean) && isset($this->properties[$property])) {
-                $boolean = $this->properties[$property];
-            }
-
-            $filterBoolean = filter_var($boolean, FILTER_VALIDATE_BOOLEAN, FILTER_NULL_ON_FAILURE);
+            if (in_array($value, ['true', '1'], true)) {
+                $value = true;
+            } elseif (in_array($value, ['false', '0'], true)) {
+                $value = false;
+            } else {
+                if (null !== $this->logger) {
```

Comment:
> That said, this implies we must check that if disabled, Symfony will inject a NullLogger instance instead of not providing any instance at all which would make everything fail.

Hmm... Perhaps an `api_platform.logger` alias?